frontend/packages/dev-console/src/components/import/deployImage-submit-utils.ts

```diff
@@ -95,28 +95,51 @@ const getMetadata = (formData: DeployImageFormData) => {
 
 export const createDeployment = (
   formData: DeployImageFormData,
-  imageStreamName: string,
   dryRun: boolean,
 ): Promise<K8sResourceKind> => {
   const {
+    registry,
     project: { name: namespace },
     name,
-    isi: { ports },
+    isi: { image, ports },
     deployment: { env, replicas },
     labels: userLabels,
     limits: { cpu, memory },
+    imageStream: { image: imgName, namespace: imgNamespace, tag },
   } = formData;
 
+  const defaultAnnotations = {
+    ...annotations,
+    ...{
+      'alpha.image.policy.openshift.io/resolve-names': '*',
+      'image.openshift.io/triggers': JSON.stringify([
+        {
+          from: {
+            kind: 'ImageStreamTag',
+            name: `${imgName || name}:${tag}`,
+            namespace: imgNamespace || namespace,
+          },
+          fieldPath: `spec.template.spec.containers[?(@.name=="${name}")].image`,
+        },
+      ]),
+    },
+  };
+
   const { labels, podLabels, volumes, volumeMounts } = getMetadata(formData);
 
+  const imageRef =
+    registry === registryType.External
+      ? `${imgName || name}:${tag}`
+      : _.get(image, 'dockerImageReference');
+
   const deployment = {
     kind: 'Deployment',
     apiVersion: 'apps/v1',
     metadata: {
       name,
       namespace,
       labels,
-      annotations,
+      defaultAnnotations,
```

Comment:
Is this really supposed to be `defaultAnnotations`? 

I see the data going up but the echo back from the server doesn't have this property in the metadata... I assume you want to be adding actual annotations?

Reply:
Ah - yeah the import utils work fine, this appears to be a bug.

Reply:
good catch. it should be `annotations: defaultAnnotations`. Changed.